sdk/python/src/dagger/mod/_converter.py

```diff
@@ -59,24 +61,29 @@ def dagger_type_unstructure(obj):
     return conv
 
 
+@functools.cache
```

Comment:
Mostly just curious, how does this differentiate different inputs for `annotation: type`? i.e. does it consider each instance of the object to be distinct vs. does it do some sort of "deep equals"-like logic on the value of the object?

Reply:
It's based on [hash](https://docs.python.org/3/library/functions.html?highlight=hash#hash):
```python
>>> to_typedef(str | None)
<dagger.TypeDef object at 0x1029032c0>
>>> to_typedef(None | str)
<dagger.TypeDef object at 0x1029032c0>
>>> to_typedef.cache_info()
CacheInfo(hits=1, misses=1, maxsize=None, currsize=1)
>>> hash(str | None)
9088350826229485722
>>> hash(None | str)
9088350826229485722
```
The argument `annotation` should only be types and they are stable and deterministic. If for some reason it isn't, it'll just cache miss but still execute.